blog/introducing-api-history.md

```diff
@@ -0,0 +1,268 @@
+> are necessary if the change is backported to other branches."
+>
+> — David Sanders [(@dsanders11)][dsanders11] via Slack
+
+We also didn't include removals in the API History, since when an API is removed
+from Electron, it is also removed from the documentation.
+
+#### JavaScript details
+
+I originally planned to create a new `@electron/docs-api-history-tools`
+NPM package that would contain scripts for extracting, validating/linting and converting
+the API history in the documentation files.
+
+About a week before the coding period began, and after some discussion with my
+mentors, I realized that was probably unnecessary:
+
+> "Hi everyone, I was thinking about the project after our huddle: Considering
+> that extraction logic will have to be handled differently in `e/website` and
+> `e/lint-roller` because of their dependencies, maybe there is no need for a
+> separate package for API history stuff?"
+>
+> |         Proposed         |        Revised         |
+> | :----------------------: | :--------------------: |
+> | ![proposed][js-proposed] | ![revised][js-revised] |
+>
+> — Piotr Płaczek (me) via Slack
+
+We ultimately decided to not go ahead with my original idea:
+
+> "@Piotr Płaczek that seems fine to me! I think we can always refactor out to a
+> separate module in a later iteration if we find that we need to share a lot of
+> code between the two implementations anyways :slightly_smiling_face:"
+>
+> — Erick Zhao ([@erickzhao][erickzhao]) via Slack
+
+Instead, we divided those various tools across the Electron repos that were most
+relevant to them:
+
+- `yaml-api-history-schema.json`
+  - -> `electron/electron` (`api-history.schema.json`)
+- `lint-yaml-api-history.ts`
+  - -> `electron/lint-roller` (`lint-markdown-api-history.ts`)
+- `extract-yaml-api-history.ts`
+  - -> `electron/website` (`preprocess-api-history.ts`)
+- `yaml-api-history-to-markdown.ts`
+  - -> `electron/website` (`transformers/api-history.ts`)
+  - -> `electron/website` (`ApiHistoryTable.tsx`)
+
+#### UI and styling for Electron documentation website
+
+I originally proposed a simple table to display the API History data:
+
+|      Design Prototype (Closed)      |      Design Prototype (Open)      |
+| :---------------------------------: | :-------------------------------: |
+| ![demo1][prototype-closed-proposed] | ![demo2][prototype-open-proposed] |
+
+This is what the final implemented design looks like:
+
+![demo3][open-implemented]
+
+Pretty much the same as the prototype. The most significant addition is the use
+of [SemVer](https://semver.org/) ranges, which were chosen to better communicate
+which versions a feature is present in (thanks Samuel Attard
+[(@MarshallOfSound)](https://github.com/MarshallOfSound) for the suggestion!).
+
+#### Usage/style Guide
+
+I added a usage/style guide dedicated to writing API history documentation for
+new features. I described proper usages of the YAML schema in detail, provided
+typical/useful examples, etc.
```

Comment:
Let's make sure to link/refer to the style guide change in the final publish: https://github.com/electron/electron/pull/42982/files#diff-7fff3714ca300b7e72b8eb5523f1dcf71fc36d031af7c1b019aeab84d6b017b7

Reply:
You mean something like this?

```patch
diff --git a/blog/introducing-api-history.md b/blog/introducing-api-history.md
index 8a9b6eb..410d8b9 100644
--- a/blog/introducing-api-history.md
+++ b/blog/introducing-api-history.md
@@ -151,7 +151,8 @@ which versions a feature is present in (thanks Samuel Attard
 
 I added a usage/style guide dedicated to writing API history documentation for
 new features. I described proper usages of the YAML schema in detail, provided
-typical/useful examples, etc.
+typical/useful examples, etc. You can find it
+[here](https://github.com/electron/electron/blob/main/docs/styleguide.md#api-history).
 
 ### Migration Guide
 

```

Reply:
Yep, although we should probably link the relative path so that we link the web page on Docusaurus build.

Reply:
CI will fail since it's not there yet, do you have an ETA on when it will? 😅

Reply:
Resolved in [`d3ca563` (#627)](https://github.com/electron/website/pull/627/commits/d3ca563b5bcc62270fe2387c2e5f1bdd61faa02c)

Reply:
Reverted in [`1c08c92` (#627)](https://github.com/electron/website/pull/627/commits/1c08c92ab6a1b133a1b9a9b8fe6cdf655c435401), see Slack.